[ShapProp] Refactor TensorMeta to use qparams for more general quantization support

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65287
* __->__ #64787

Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

jerryzh168 committed:
[fx2trt][quant] Add lowering support for per channel quantization in fx2trt

Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/fx/passes/shape_prop.py:
```diff
@@ -21,6 +21,9 @@ class TensorMetadata(NamedTuple):
     qscheme : Optional[torch.qscheme]
     q_scale : Optional[float]
     q_zero_point : Optional[int]
+    q_per_channel_scales : Optional[torch.Tensor]
```
Hi @jamesr66a, it looks like TensorMeta needs to be backward compatible, how can we add new things to it?

jamesr66a replied:
Add your class member with a default value that's assigned in the body of the class: https://docs.python.org/3/library/typing.html#typing.NamedTuple

After you do that, `--accept` test_class_member_back_compat and tag me for review

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

BC test for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

842974287 commented on torch/fx/experimental/fx_acc/acc_ops.py:
```diff
@@ -389,30 +388,26 @@ def min_single_tensor_input(*, input):
-)
 @register_acc_op
-def quantize_per_tensor(*, input, acc_out_ty=None):
+def quantize_per_tensor(*, input, qparams=None):
```
We might still want to use `acc_out_ty` here to be consistent with other quantized ops. Instead we can change how `kwargs_to_move_to_acc_out_ty` works so it would create `qparams` correctly.

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented on torch/fx/experimental/fx2trt/converters/acc_ops_converters.py:
```diff
@@ -1367,23 +1408,35 @@ def acc_ops_dequantize(network, target, args, kwargs, name):
+        scale_shape = q_scale.shape
+        if not torch.equal(q_zero_point, torch.zeros(q_zero_point.shape, dtype=q_zero_point.dtype)):
+            raise RuntimeError(f"Only support zero_point == 0, get {q_zero_point}")
+
```
yeah sure

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

842974287 commented on torch/fx/experimental/fx_acc/acc_ops.py:
```diff
@@ -359,6 +358,51 @@ def matmul(*, input, other):
     return torch.matmul(**locals())
 
 
+@register_custom_acc_mapper_fn(
```
Remove? Seems like something wrong happened during rebasing.

jerryzh168 replied:
oh right

842974287 commented on torch/fx/experimental/fx_acc/acc_normalizer.py:
```diff
@@ -60,6 +60,9 @@ class NormalizationInfo(NamedTuple):
     needs_shapes_for_normalization: bool
+    # qscheme of Tensor, used to get different fields for quantization
+    # parameters
+    qscheme: Optional[torch.qscheme]
```
`qscheme` doesn't feel like a good fit in `NormalizationInfo`. Why do we need this? I saw it mostly just hard coded for some ops.

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

842974287 commented on torch/fx/experimental/fx_acc/acc_normalizer.py:
```diff
@@ -243,26 +235,38 @@ def move_kwargs_to_acc_out_ty(
+            orig_kwarg_name, tmd_field_name, is_constant = kwarg_replacement_tuple
+        if is_constant:
+            # when the arg is constant, the fields will be name and value
+            tmd_dict[orig_kwarg_name] = tmd_field_name
```
Can we make the tuple (constant, tmd_field_name, is_constant) so that here we can do `tmd_dict[tmd_field_name) = orig_kwarg_name`?

jerryzh168 replied:
oh OK

842974287 reviewed (approved):
Looks good to me! Just one more comment and cc: @jfix71 to take another look at the changes on `kwargs_to_move_to_acc_out_ty`.

jamesr66a reviewed (commented):
Can you separate out the ShapeProp changes into a separate PR? That will make it easier to keep track of the origin of BC-breaking changes

jerryzh168 committed:
Update on "[fx2trt][quant] Add lowering support for per channel quantization in fx2trt"


Summary:
This PR added support for lowering per channel quantization and dequantization operators
in fx2trt, this also extends TensorMeta with extra arguments corresponding to per channel quantized Tensors,
initially I was thinking of adding a qpram that can capture everything, but currently we still have some lowering support
for fbgemm ops (which has scale and zero_point in operator interface). I think we can move everything to qprams
after we deprecate lowering support for fbgemm ops in the future.

Test Plan:
Test for per channel weight:
```
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py
```

change BC compatibility test expect for TensorMeta
```
python test/test_fx.py TestFXAPIBackwardCompatibility.test_class_member_back_compat --accept
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
> Can you separate out the ShapeProp changes into a separate PR? That will make it easier to keep track of the origin of BC-breaking changes

sure, moved out the quantize_per_channel changes to a separate PR, please take a look again

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jamesr66a commented on torch/fx/passes/shape_prop.py:
```diff
@@ -45,20 +43,27 @@ def _extract_tensor_metadata(result : torch.Tensor) -> TensorMetadata:
-
+        qparams = {}
+        qparams["qscheme"] = result.qscheme()
+        qparams["dtype"] = result.dtype  # type: ignore[assignment]
```
What error are these `# type: ignore` comments suppressing? Can we type qparams as `Dict[str, Any]` and make this go away?

jamesr66a reviewed (approved):
This looks fine to me. I suggest also requesting review from the owners of the other components that are being changed (fx2trt, ACC tracer, graph drawer) and make sure those changes are OK

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[ShapProp] Refactor TensorMeta to use qparams for more general quantization support"


Summary:
This PR refactors TensorMeta to use qparams dictionary which will be scalable for
all future quantization schemes, this will be a bc-breaking change for TensorMeta,
but since bc-guarantee has not been checkedin for 1.9 we can still do it. This PR should
be landed before 1.10 branch cut.

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30879848](https://our.internmc.facebook.com/intern/diff/D30879848)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30879848).
